integration/users/create_user_test.go

```diff
@@ -125,6 +152,10 @@ func TestCreateUser(t *testing.T) {
 	for name, tc := range testCases {
 		name, tc := name, tc
 		t.Run(name, func(t *testing.T) {
+			if tc.hasCred && setup.IsMongoDB(t) {
+				t.Skip("PLAIN credentials are only supported via LAP (PLAIN) by MongoDB Enterprise")
```

Comment:
```suggestion
				t.Skip("PLAIN credentials are only supported via LDAP (PLAIN) by MongoDB Enterprise")
```

Reply:
In general, we should prefer expected failures to just skips

Reply:
Should I replace this with something like the following?

```go
t.Run(name, func(tt *testing.T) {
	tt.Parallel()

	var t testtb.TB = tt
	if tc.hasCred && setup.IsMongoDB(t) {
		t = testfail.Expected(t, "PLAIN credentials are only supported via LDAP (PLAIN) by MongoDB Enterprise")
	}

	// ...
}
```

I'm asking because I haven't seen this pattern anywhere and it seems we don't have an analogue to `setup.FailsForMongoDB` to when we expect a test run with MongoDB to fail.

Reply:
> and it seems we don't have an analogue to `setup.FailsForMongoDB`

So let's make one in the next PR

Reply:
Okay! I thought it didn't exist on purpose. Should've have asked! 🤣